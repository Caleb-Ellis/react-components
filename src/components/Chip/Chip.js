import React from "react";
import PropTypes from "prop-types";

import "./Chip.scss";

const Chip = ({ value }) => {
  return (
    <div className="p-chip">
      { value }
    </div>
  );
};

Chip.propTypes = {
  value: PropTypes.string,
};

export default Chip;
