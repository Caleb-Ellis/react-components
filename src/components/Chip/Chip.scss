@import "~vanilla-framework/scss/base_placeholders";
@include vf-b-placeholders;

.p-chip {
  @extend %small-text;

  background-color: rgba($color-x-dark, 0.05);
  border-radius: 1rem;
  display: inline-block;
  margin-bottom: 0;
  margin-right: $sph-inner--small;
  padding: 0 $sph-inner map-get($nudges, nudge--small);
  white-space: nowrap;
  user-select: none;

  &:hover {
    background-color: rgba($color-x-dark, 0.08);
  }

  &.is-selected {
    background-color: rgba($color-x-dark, 0.2);

    &:hover {
      background-color: rgba($color-x-dark, 0.23);
    }
  }

  &:active {
    background-color: rgba($color-x-dark, 0.2);
  }
}
