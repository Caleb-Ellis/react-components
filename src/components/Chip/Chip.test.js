import { shallow, mount } from "enzyme";
import React from "react";

import Chip from "./Chip";

describe("Chip ", () => {
  it("renders", () => {
    const wrapper = shallow(<Chip value="positive" />);
    expect(wrapper).toMatchSnapshot();
  });

  it("displays the value", () => {
    const wrapper = mount(<Chip value="positive" />);
    expect(wrapper.find(".p-chip").text()).toBe("positive");
  });
});