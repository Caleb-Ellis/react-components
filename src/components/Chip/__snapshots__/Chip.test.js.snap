// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip  renders 1`] = `
<div
  className="p-chip"
>
  positive
</div>
`;
